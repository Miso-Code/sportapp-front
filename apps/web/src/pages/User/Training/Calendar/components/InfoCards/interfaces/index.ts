import { ReactNode } from 'react'

export enum StateTypes {
	BACK = 'default',
	TODAY = 'primary',
	NEXT = 'warning'
}

export interface Props {
	titleComponent: string | ReactNode
	description: string | ReactNode
	state: { type: `${keyof typeof StateTypes}`; text: string }
	date: string
	showChip?: boolean
	onClick: () => void
}
