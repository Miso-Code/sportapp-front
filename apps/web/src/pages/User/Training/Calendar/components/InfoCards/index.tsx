import {
	Box,
	Card,
	CardActionArea,
	CardContent,
	Chip,
	Typography
} from '@mui/material'
import { Props, StateTypes } from './interfaces'

export default function InfoCards({
	titleComponent,
	description,
	date,
	state,
	onClick,
	showChip = true
}: Props) {
	return (
		<Card>
			<CardActionArea onClick={onClick}>
				<CardContent>
					<Box
						sx={{
							display: 'flex',
							justifyContent: 'space-between',
							pb: 2
						}}>
						<Typography variant='body1' color='GrayText'>
							{date}
						</Typography>
						{showChip && (
							<Chip
								variant='filled'
								color={StateTypes[state.type]}
								label={state.text}
							/>
						)}
					</Box>
					<Typography gutterBottom variant='h6'>
						{titleComponent}
					</Typography>
					<Typography
						variant='body2'
						color='GrayText'
						component='div'>
						{description}
					</Typography>
				</CardContent>
			</CardActionArea>
		</Card>
	)
}
