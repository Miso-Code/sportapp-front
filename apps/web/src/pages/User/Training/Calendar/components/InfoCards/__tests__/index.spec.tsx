import InfoCards from '..'
import { render, RenderResult } from '@testing-library/react'

describe('InfoCards', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<InfoCards
				titleComponent='Test'
				description='Test'
				state={{ type: 'BACK', text: 'Test' }}
				date='Test'
				onClick={() => {}}
			/>
		)
	})

	afterEach(() => {
		wrapper.unmount()
	})

	it('should render', () => {
		expect(wrapper.container).toMatchSnapshot()
	})

	it('should render with showChip', () => {
		wrapper.rerender(
			<InfoCards
				titleComponent='Test'
				description='Test'
				state={{ type: 'BACK', text: 'Test' }}
				date='Test'
				onClick={() => {}}
				showChip={false}
			/>
		)
		expect(wrapper.container).toMatchSnapshot()
	})
})
