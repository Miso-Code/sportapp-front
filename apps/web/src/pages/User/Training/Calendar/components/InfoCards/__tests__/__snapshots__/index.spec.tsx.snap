// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCards should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1jyfzin"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1m8eymn-MuiTypography-root"
          >
            Test
          </p>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Test
            </span>
          </div>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Test
        </h6>
        <div
          class="MuiTypography-root MuiTypography-body2 css-x6zrxh-MuiTypography-root"
        >
          Test
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`InfoCards should render with showChip 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1jyfzin"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1m8eymn-MuiTypography-root"
          >
            Test
          </p>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Test
        </h6>
        <div
          class="MuiTypography-root MuiTypography-body2 css-x6zrxh-MuiTypography-root"
        >
          Test
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
